config-model/src/main/java/com/yahoo/vespa/model/container/component/ConfigProducerGroup.java

```diff
@@ -51,14 +51,15 @@ public Collection<CHILD> getComponents() {
     }
 
     public <T extends CHILD> Collection<T> getComponents(Class<T> componentClass) {
-        Collection<T> result = new ArrayList<>();
+        List<T> result = new ArrayList<>();
 
         for (CHILD child: getChildren().values()) {
             if (componentClass.isInstance(child)) {
                 result.add(componentClass.cast(child));
             }
         }
-        return Collections.unmodifiableCollection(result);
+        // We need consistent ordering
+        return result.stream().sorted().toList();
```

Comment:
Since you are introducing streams here I would go all the way with
```
return getChildren().values().stream()
                .filter(componentClass::isInstance)
                .map(componentClass::cast)
                .sorted().toList();
```

Reply:
Yeah, considered that, but did the least possible change. Probably better to go all the way, as you say.